bumpversion/vcs.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals, print_function
+
+import logging
+import os
+import subprocess
+from tempfile import NamedTemporaryFile
+
+from bumpversion.exceptions import WorkingDirectoryIsDirtyException, MercurialDoesNotSupportSignedTagsException
+from bumpversion.compat import _command_args
+
+
+logger = logging.getLogger(__name__)
+
+
+class BaseVCS(object):
+    @classmethod
+    def commit(cls, message):
+        f = NamedTemporaryFile("wb", delete=False)
+        f.write(message.encode("utf-8"))
+        f.close()
+        env = os.environ.copy()
+        env[str("HGENCODING")] = str("utf-8")
+        try:
+            subprocess.check_output(cls._COMMIT_COMMAND + [f.name], env=env)
+        except subprocess.CalledProcessError as exc:
+            err_msg = "Failed to run {}: return code {}, output: {}".format(
+                exc.cmd, exc.returncode, exc.output
+            )
+            logger.exception(err_msg)
+            raise exc
+        os.unlink(f.name)
+
+    @classmethod
+    def is_usable(cls):
+        try:
+            return (
+                subprocess.call(
+                    cls._TEST_USABLE_COMMAND,
+                    stderr=subprocess.PIPE,
+                    stdout=subprocess.PIPE,
+                )
+                == 0
```

Comment:
This is, at least to me, a very uncommon coding style and doesn't look very readable. Could we use `subprocess.check_call` instead?